Firmware/MotorControl/thermistor.cpp

```diff
@@ -0,0 +1,86 @@
+#include "odrive_main.h"
+
+#include "low_level.h"
+
+ThermistorCurrentLimiter::ThermistorCurrentLimiter(uint16_t adc_channel,
+                                                   const float* const coefficients,
+                                                   size_t num_coeffs,
+                                                   const float& temp_limit_lower,
+                                                   const float& temp_limit_upper,
+                                                   const bool& enabled) :
+    adc_channel_(adc_channel),
+    coefficients_(coefficients),
+    num_coeffs_(num_coeffs),
+    temp_(),
+    temp_limit_lower_(temp_limit_lower),
+    temp_limit_upper_(temp_limit_upper),
+    enabled_(enabled),
+    error_(ERROR_NONE)
+{
+}
+
+float ThermistorCurrentLimiter::get_temp() const {
+    return temp_;
+}
+
+void ThermistorCurrentLimiter::update() {
+    const float voltage = get_adc_voltage_channel(adc_channel_);
+    const float normalized_voltage = voltage / adc_full_scale;
+    temp_ = horner_fma(normalized_voltage, coefficients_, num_coeffs_);
+}
+
+bool ThermistorCurrentLimiter::do_checks() {
+    if (enabled_ && temp_ >= temp_limit_upper_) {
```

Comment:
We should keep the 5°C margin here that was in the old code to account for measurement noise.

Reply:
Fixed